### ISSUE TITLE
iOS app

### ISSUE THREAD

**Issue by binwiederhier** [binwiederhier/ntfy#4]
If you're interested in making an iOS app, I'm more than happy to help out. Unfortunately I don't have an iPhone or a Mac so it's a little tricky for me.

**Comment by Copephobia** [comment]
I am working on the iOS app for ntfy, in case github cannot assign the issue to me.

**Comment by binwiederhier** [comment]
Hey @Copephobia, can you give an update of how it's going? Need any help?

**Comment by gedw99** [comment]
Hey you can build cross platform golang apps using gio 

It has notifications plugin here:
https://github.com/gioui/gio-x/tree/main/notify

i use it . It is easy no works . Not web notification yet but quite easy with gio - just look at other plugins to see how .

gio is all golang and compiles for web , all desktops, all mobiles.

just wanted to highlight this as an option .

mug you search for “gioui” you will find many apps built with it

**Comment by binwiederhier** [comment]
Sadly I haven't heard from @Copephobia in weeks, so I don't think it'll happen :-( 

I'm gonna go looking for an iOS dev.

**Comment by gedw99** [comment]
I read on the Unified Push web site that the IOS app needs to support Background worker , but i don’t think it’s possible . AppNap on iOS puts applications into sleep mode .

gio android does support foreground and background btw .

gio iOS only supports foreground and so the IOS AppNap never occurs.

**Comment by binwiederhier** [comment]
For the iOS app I'd love at least basic messages with firebase for ntfy.sh. i think that should be possible hopefully. Whether or not and to what extent other features are supported we'll have to see. But for now I can't even find anyone to do the iOS basic app.

**Comment by gedw99** [comment]
well you could do it with golang, but  GIo does not have a IOS driver yet. It does have a driver for  web, desktop ( win, mac, linux ) and mobile (android). I think that it could be done in a few days.

https://github.com/gioui/gio-x/tree/main/notify

Looking for devs using it: https://github.com/search?o=desc&p=2&q=gioui.org%2Fx%2Fnotify&s=indexed&type=Code

found one decent example: https://github.com/arborchat/sprig/blob/main/core/notification-service.go

**Comment by Copephobia** [comment]
Current work towards iOS, feel free to follow along: https://github.com/Copephobia/ntfy-ios

(Relevant macOS development branch: https://github.com/Copephobia/ntfy/tree/macos-development)

**Comment by binwiederhier** [comment]
For transparency, here's the contents of the Discord #ios channel. That room is now bridged to Matrix as well: https://matrix.to/#/#ntfy-ios:matrix.org

![220207 14-50-09 Selection 001](https://user-images.githubusercontent.com/664597/152869496-bf3186e5-e55e-4ae5-b2f1-29a7a992c243.png)

![220207 14-50-20 Selection 001](https://user-images.githubusercontent.com/664597/152869507-aac470fa-ba73-4e95-bdc7-fff03620b6dc.png)

![220207 14-50-31 Selection 001](https://user-images.githubusercontent.com/664597/152869518-10dfdaa0-a1e4-4739-85bf-15bfcf717cc9.png)

![220207 14-50-42 Selection 001](https://user-images.githubusercontent.com/664597/152869533-39c75bc5-6fa4-4acb-bf53-115d65d575e1.png)

![220207 14-50-53 Selection 001](https://user-images.githubusercontent.com/664597/152869557-56013bd7-4471-4bd6-94df-4325b5b70096.png)

**Comment by easrng** [comment]
I don't know if you can set it to automatically check for notifications but [here's](https://www.icloud.com/shortcuts/0436305ef0424526bc1027d3068a01ef) an iOS Shortcut that checks for new ntfy notifications and shows them. Unfortunately you can't have a Shortcut run on a fixed interval, only on manually added times (ex. you can set a Shortcut to run every day at 2pm, but not every 5 minutes) so if you want it to check automatically you're going to be adding a LOT of automations.

**Comment by lylebrown** [comment]
> [Here's](https://www.icloud.com/shortcuts/0436305ef0424526bc1027d3068a01ef) an iOS Shortcut that checks for new ntfy notifications and shows them. Unfortunately you can't have a Shortcut run on a fixed interval, only on manually added times (ex. you can set a Shortcut to run every day at 2pm, but not every 5 minutes) so if you want it to check automatically you're going to be adding a LOT of automations. It supports titles, messages, and turns off sound for low-priority notifications.

I would love to be proven incorrect, but doesn't iOS require the device to be unlocked (as in, the screen turned on and not on the lock screen) for shortcuts to run?

**Comment by Seji64** [comment]
Hello,
happened to come across this issue as I would like to use ntfy on iOS.

So I took a look at the iOS code ([https://github.com/Copephobia/ntfy-ios](https://github.com/Copephobia/ntfy-ios)). As I see it, the app is practically ready. I think the problem why the development was stopped is that Copephobia probably didn't find a good solution for the notifications. I thought about the problem and wondered how Mastadon solved it.

Because here each instance is self-hosted and yet the notification works via a central official app. I came across the various proxy's / relay's (such as [https://github.com/DagAgren/toot-relay](https://github.com/DagAgren/toot-relay)). I think this could solve the whole thing quite elegantl. Sorry if I now come here with an 'idea' that was already known to all of you - did not want to be the wise guy - just give a hint or help.

A good explanation how this all works i found [here](https://github.com/Big-Fig/toot-relay-fcm#how-it-works=)

is ntfy already capable to send such notification to a push relay?

**Comment by binwiederhier** [comment]
Thank you for looking at the code! I am actually beginning to see if I can borrow and get a Mac from somewhere so I can learn Swift and all that myself. But I'm always happy to have someone else look or take over. I bet that'd be much quicker than me learning how to use a Mac, iOS, and Swift :-D

So @Copephobia and I actually found a solution already for this problem. We talked about a `poll_request` message over FCM, which basically just instructs the phone to poll from a selfhosted server. There are some notes about this in the (now closed) pull request (https://github.com/binwiederhier/ntfy/pull/152). I'll look at the relay/proxy you linked though. Maybe they had a better idea :-D

**Comment by Seji64** [comment]
I have a mac - so if i can help out - let me know. The problem is that I also have to learn swift ... :-/

**Comment by binwiederhier** [comment]
@Seji64 If you wanna give it a shot I'm not gonna say no. :-D But I also don't blame you if you didn't want to. It's a lot of work.

**Comment by binwiederhier** [comment]
My company has loaned me an old 2017 Macbook. After the next Android and server release, I am going to give the iOS app a go.

**Comment by binwiederhier** [comment]
Let's do this!

![IMG_20220505_225513](https://user-images.githubusercontent.com/664597/167059427-fef211b6-a7d7-4611-8beb-a1e01fb8eae4.jpg)

**Comment by binwiederhier** [comment]
Here's what I got so far. This is 99% @Copephobia's work. All I did was remove stuff in an attempt to ship something soon. Firebase does not work in the simulator (or so I've heard), so I have to acquire an iPhone to test the entire flow.

It also looks not very great IMHO, though the iPhone apps apparently look like this.

https://user-images.githubusercontent.com/664597/167267068-d80a98e8-c184-48fc-a7f9-9059e9c59da5.mov

**Comment by Seji64** [comment]
Wow, this looks great! Any kind of relay for push notifications are not yet implemented ?

**Comment by binwiederhier** [comment]
It needs a physical iOS device and i don't have one yet. I'm asking all the people if they have old/spare iPhone devices so I can test. I'll likely just buy an iPhone 11 though. It's "only" $500...

**Comment by binwiederhier** [comment]
> Wow, this looks great!

Does it really "look great"? I am legit lost as to the UI design in iOS. I haven't really looked too closely, but to me iOS apps (from Googling screenshots) look uuuuuggggllly. That's part of the reason why I want a real iPhone, so I can see what popular apps look like.

**Comment by fosspill** [comment]
> > Wow, this looks great!
> 
> Does it really "look great"? I am legit lost as to the UI design in iOS. I haven't really looked too closely, but to me iOS apps (from Googling screenshots) look uuuuuggggllly. That's part of the reason why I want a real iPhone, so I can see what popular apps look like.

Honestly looks pretty good to me too!
If there is anything I could record or screenshot for you to see, let me know

**Comment by binwiederhier** [comment]
Screenshots and/or videos of messaging apps like WhatsApp or whatever else iOS people use would be helpful indeed.

**Comment by Seji64** [comment]
>  Screenshots and/or videos of messaging apps like WhatsApp or whatever else iOS people use would be helpful indeed.

here we go...

### Signal

![Singal01](https://user-images.githubusercontent.com/19726119/167304934-b13d0036-637c-46b4-8680-cc02a432e97c.jpeg)

### WhatsApp
![WhatsApp](https://user-images.githubusercontent.com/19726119/167304856-5fcb5f10-1981-4ec3-81a6-cfa68b27e470.jpeg)

### Mastadon

![IMG_8ED1E2C4E17B-1](https://user-images.githubusercontent.com/19726119/167304989-e8f96753-f2e9-4dfe-9a0e-796627fe6553.jpeg)

**Comment by binwiederhier** [comment]
I'm getting my loaner iPhone today, so I should be able to make some progress from now on.

(Cross post since y'all are Apple fans: This ticket (https://github.com/binwiederhier/ntfy/issues/112#issuecomment-1123106697) has a macOS build of the `ntfy` binary (client only, no server), in case people want to try that out.)

**Comment by binwiederhier** [comment]
I've been poking around and trying out the Apple notification system, and it seems severely limited, compared to Android. Even Android seemed limited at the time, but iOS is almost unusable.

- It seems impossible to run code when you receive a notification
- APNS/Apple handles displaying the notification and only when the user interacts with it, hooks can be run
- I think I can make this work for ntfy.sh, but it is entirely questionable if a selfhosted instance can be made compatible with this.
- My original idea to implement a poll_request that we forward via ntfy.sh/firebase (selfhosted server ---(poll_request)---> ntfy.sh -> firebase -> apns -> ntfy iphone app --(poll)--> selfhosted server) seems impossible now
- Simply encrypting the message and using ntfy.sh/firebase/apns is also impossible, because we cannot run code to decrypt. 

I may be overlooking something, because I have not fully grasped the notification extension that @Copephobia implemented (https://developer.apple.com/documentation/usernotifications/unnotificationserviceextension). I do not see any evidence that it runs, and it does look promising. This flow may be doable if this works: selfhosted server ---(encrypted notification)--->ntfy.sh-->firebase-->apns-->iphone app (decrypts)

Still at the beginning of this. My guess is that it will take multiple weeks for me to fully grasp this and be able to publish in the app store. Unless someone with experience comes along.

Cheerio.

**Comment by spantaleev** [comment]
Not sure how [element-ios](https://github.com/vector-im/element-ios) does it, but from what I know, notifications there are just a wakup call. After being woken up, the application then retrieves the new message from a self-hosted homeserver, decrypts it (if End-to-End-Encrypted) and renders the notification text by itself.

**Comment by prabirshrestha** [comment]
This is a good read from on how simplex chat iOS notifications work. https://github.com/simplex-chat/simplex-chat/blob/stable/blog/20220404-simplex-chat-instant-notifications.md

**Comment by binwiederhier** [comment]
Thank you for the links. I read the one about simplexchat, and it sounds like I've understood the Apple APIs correctly. 

Tldr: There are no way to deliver instantly without APNS, meaning a central server. Periodic checks apparently stop working after a while all together, so that's cool..  

I think I'm on the right track with the notification service extension. I'll check out the element code too.

**Comment by prabirshrestha** [comment]
A good compromise could be to add additional privacy option similar to priority and tags. If I don’t care about privacy I can change it so it doesn’t need to be encrypted. This would allow me to send messages such as:

- “A new event occurred on you NAS.”
-  “new update available in your router.”
- “User logged in via ssh”.

I could then send another follow up message with more details that is private if needed.

Messages could be private by default to enable backwards compatibility.

**Comment by binwiederhier** [comment]
I am essentially at a stand-still with this. I cannot get the Notification Service Extension to work. I've Googled and tried many many things, but all the solutions do not work. I've tried creating a new project, I've followed tutorials, none of it works. The code that is supposed to modify the notification is just not called. 

I'll keep trying, but this is supremely frustrating.

**Comment by binwiederhier** [comment]
I've started a new project in Xcode and watched tons of tutorials and read tons of documentation pages and I figured out the above problem.

I think I've truly turned a corner. It's turned from "what the heck" to "okay, I think I can do this in a couple of days". The hard pieces are clear to me now: Message delivery with Firebase, Views/layouting and persistence with live-updating views. I'm going to be deviating from @Copephobia's app a little in that I'll be using CoreData on top of SQLite for persistence, since that makes live-updating views super easy.

As for self-hosted servers, I believe ultimately, these two options should be possible (with option 1 being significantly easier I think):

1. ntfy.sh/firebase (selfhosted server ---(poll_request)---> ntfy.sh -> firebase -> apns -> ntfy iphone app --(poll)--> selfhosted server) 
2. selfhosted server ---(encrypted notification)--->ntfy.sh-->firebase-->apns-->iphone app (decrypts)

--

I have started from a completely fresh Xcode project, and I am pulling in whatever I need from the code in Cope's repo. You can follow my progress here: https://github.com/binwiederhier/ntfy-ios/tree/restart. Right now it doesn't look like much, but as I said, I believe I've figured out the hardest pieces.

**Comment by binwiederhier** [comment]
Lots of progress: https://www.icloud.com/photos/#044Nwj6y2ALrcfToCc64hVKJA

**Comment by binwiederhier** [comment]
Good progress again with iOS today. I don't really understand why I had to make the hacks that I did, but it works (on my device). I'm going to polish a lot the next few days, and then I should be able to release soon.

The initial release will only support ntfy.sh (no selfhosted) and only plain messages with/without title. No tags, attachments, click URLs, ...

**Comment by binwiederhier** [comment]
I think the iOS app is ready for testing, once I manage to upload it. I gotta do a tiny server release before though. Will do that tomorrow.

**Comment by binwiederhier** [comment]
https://photos.app.goo.gl/7Uu2yEPfRVC7Du4z6

**Comment by binwiederhier** [comment]
Now I'm waiting for Apple. :apple: I believe that if you are interested, I can invite you as an internal tester already. Message me on Discord or Matrix.

![image](https://user-images.githubusercontent.com/664597/169667911-7cf31e4f-be47-4edc-ac4c-8acedcc56734.png)

**Comment by binwiederhier** [comment]
# :loudspeaker: :loudspeaker: :loudspeaker: ntfy iOS beta testing

Please test via TestFlight and give me feedback here or via TestFlight itself. Thank you!

### Open this on your iPhone/iPad: https://testflight.apple.com/join/P1fFnAm9

**Comment by fosspill** [comment]
Thank you for your hard work!

My test results:
- No messages are received in foreground and background. If I manually refresh (pulling down inside the app) I do receive the messages, however.
- UI seems to respond correctly when I refresh to see the notifications
- After unsubscribing I no longer get anything when I refresh (by pulling down, again)

**Comment by binwiederhier** [comment]
Well that's a bummer. That means Firebase is not working, or the app extension is not working. I don't quite know how to debug things, because the logs are not written to a file... What device is this?

**Comment by fosspill** [comment]
iPhone 12. I submitted feedback through testflight too, which I think should contain logs and such?

**Comment by prabirshrestha** [comment]
Works for me on both iPad and iPhone whether an app is running or not or is in background or foreground.

I already have bunch of feature requests. Let me know if there is a better place for this.

- Simple dialog to send a message.
- First class shortcuts integration. Will easily allow Siri integration.
- copy the topic to clipboard so I can share with others or my own new device. Can be improved later by using iOS share sheet.

Looking forward to the selfhosted support.

**Comment by binwiederhier** [comment]
Please keep posting your feedback here. I will update the comment above (https://github.com/binwiederhier/ntfy/issues/4#issuecomment-1133767150) with a summary of feedback I have received so far.

**Comment by prabirshrestha** [comment]
Clicking notification should take to the notification  details page. When I had multiple notifications or wanted to copy the content I needed to find it.

Here is an example from pushover. If you enable sharesheet you don’t need to implement separate copy to clipboard button. 

![ED6A0843-0CF7-45C6-9A1D-387F7BE4E33D](https://user-images.githubusercontent.com/287744/169673693-7fc2989c-1f5d-4268-b9f1-d771e1f70346.jpeg)

“Hey Siri, notify” works. It is simple as adding 2 actions in Siri Shortcuts.

![60385464-7F38-424E-A5BC-49120A13E3ED](https://user-images.githubusercontent.com/287744/169673720-b4397aad-a18c-4350-822d-ffafda04507b.jpeg)

**Comment by fosspill** [comment]
@binwiederhier 
1. I'm unsure about exactly when I enabled notifications, but notifications were indeed enabled for ntfy during my testing. I confirmed this through settings multiple times
2. Re-installing worked, but I reinstalled on a different network in case something firebase-related was blocked during installation.
After reinstalling it also works fine on the network I originally installed the app on.

So... I'm not entirely sure about what happened here but the issue is resolved 🤷

**Comment by binwiederhier** [comment]
@fosspill Thank you for retesting. That makes me feel much better. You were the first feedback and it was "nothing works", that was quite the bummer. Glad it works now. Maybe I can somehow get an indication in the UI to show if Firebase works or not.

**Comment by fosspill** [comment]
@binwiederhier Hah, I can imagine that. It did feel pretty bad to leave that feedback right away as well, but I'm super happy that it works now. I'll finally be able to use ntfy in my workflow and that's all thanks to your hard work.
Some firebase indication would have been interesting, and I would have liked to figure out what went wrong. But, if a reinstall is all that's needed then it's not really a big issue. Maybe we can blame cosmic rays for now.

**Comment by binwiederhier** [comment]
Dont feel bad. This is exactly the feedback that I need. Only a handful from the people who try it leave any feedback. So this is good stuff. It's a really really rough version, but I want to release quickly and then iterate towards features and stability.

**Comment by skr32** [comment]
Here me feedback:
- Subscribe to different Topics
- Receive Notification on those topics. Push notification works on unlocked and locked screen and while app is in background or  closed
- Unsubscribing works. No notification for removed topics.
- Content: Title & Messages work for me

Thank you for the App!! Cant wait for Content & Tags to work ;)

**Comment by binwiederhier** [comment]
I'd like to thank everyone for the feedback. **Keep it coming!!** It helped a lot to assess the state of the app. I have decided to submit the app for review as-is by tomorrow night (Tuesday, May 24). I know there are lots of issues, but I'd rather the Apple community have something than nothing. 

Issues I am aware of:
- The UI does not always update properly (#267, ❗❗)
- When clicked, the notification does not open the topic, but instead just shows the last view
- The notification does not make a sound or vibrate on the Apple Watch
- When opening a topic, the notifications in the Apple notification center should be dismissed automatically
- Errors are not shown to the user, but instead just logged

Feature requests:
- Support for other (priority, tags, click target, actions, attachments, ...)
- Siri integration
- Copy topic to clipboard
- Copy message to clipboard
- Copy logs to clipboard (or upload)

**Comment by skrollme** [comment]
I'm new to _ntfy_ because I was looking for an (possibly self-hosted) alternative to ifttt.com push messages. As I saw the development/progress of the iOS I subscribed to this topic and tried the TestFlight build as it came out:
**👍  Works like charm and I'm already waiting for more features to come.** 
In the meantime I will switch my toolchains to use _ntfy_

**Comment by SxthGear** [comment]
A little late to the party as it looks like you're pushing live today sometime, but I just happened to see this posted on /r/selfhosted and have been waiting for iOS support for some time now. Prelim run through of different things on an iPhone 13 Pro, iOS 15.4.1:

👍 **Works perfectly, even with some odd ASCII characters and emoji thrown at it. Can't wait to run a selfhosted version!**

**Comment by binwiederhier** [comment]
Love hearing that. Thanks folks. 👍

**Comment by binwiederhier** [comment]
Rejected... For mentioning Android... I suppose that's an easy enough fix. 

"We found the metadata of your submitted in-app purchases includes irrelevant third-party platform information. Specifically, Android references is mentioned in the description.

Referencing third-party platforms in your app or its metadata is not permitted on the App Store."

**Comment by binwiederhier** [comment]
https://apps.apple.com/us/app/ntfy/id1625396347

**Comment by binwiederhier** [comment]
I'm going to keep this ticket open for a while to communicate updates on iOS, but eventually I'll close it. To stay in the iOS loop, I highly recommend joining the Discord or Matrix (links on main GIthub page).

Updates:
- Aside from attachments, I've implemented pretty much all "publish" features (priority, tags/emojis, click target, user action buttons). There is no configuration/muting/... and such as there is in Android yet, obviously.
- As for selfhosted servers, I have not started that work yet officially, BUUTT, I think I may be able to do a "selfhosted light" variant pretty quickly: Basically poll all topics every 20-60min or so (Apple is vague on this, it's also not guaranteed to run at all) by remotely triggering silent "poll now" Firebase messages. The messages are empty, they just serve as triggers to poll all topics. I've got this implemented in https://github.com/binwiederhier/ntfy-ios/pull/5/files and it works nicely. I think I can get that into the next release too, though likely without support for auth.

Here are a bunch of pictures: https://photos.app.goo.gl/KeERoCa25WqnfpuC8

**Comment by binwiederhier** [comment]
Here's a short video showing my progress on the selfhosted server support: https://photos.app.goo.gl/weSQxdbL8kkt11nS8

Here's what's happening:
- **Top screen:** User sends a message to "middle screen" (= selfhosted server)
- **Middle screen:** Selfhosted server is configured to send a message to `topicHash=sha256($topic)` containing the message ID of the original message (`poll_id=$messageID`) to an upstream server (= ntfy.sh). This is equivalent to `curl -X POST -H "Poll: $messageID" ntfy.sh/$topicHash`).
- **Bottom screen:** Upstream server (ntfy.sh) receives the poll request and forwards it to Firebase topic `$topicHash` (which forwards it to APNS). The poll request contains only the original message ID; none of the other fields of the original message. It looks like this: `{"id":"Xi6uHAal8f7i","time":1653619618,"event":"poll_request","topic":"mylongtopichash","message":"New message","poll_id":"123"}`
- **Phone:** receives the poll request from APNS, containing `$messageID`, and asks the selfhosted server for the original message (equivalent to `curl "localhost:2587/test/json?poll=1&id=Ab9dXUgm83cp`)
- Phone displays message.

--> Firebase/APNS are used as signalling mechanism, and the actual message is fetched by the phone.

![image](https://user-images.githubusercontent.com/664597/170618222-27a6facf-6d08-46d9-b946-e1d088a820b1.png)

**Comment by prabirshrestha** [comment]
You might be able to use location service feature of iOS to have frequent polling. At least that is what Dropbox and other products do. https://help.dropbox.com/installs-integrations/mobile/location-data

Other option I could see is adding shortcut to force poll so I can run it when it charges or connects to Wi-Fi and so on. Shortcuts does have show notification action.

Some more additional options here. https://stackoverflow.com/questions/35388086/is-there-a-way-to-wakeup-suspended-app-in-ios-without-user-or-server-interventio

found a bug:
If your topic is long it seems to flicker as it adjusts the size in header.

**Comment by Famku** [comment]
Hi.

I just tested NTFY App 1.1 and it works very good with self hosted ntfy.

Thank you for yor hard work!

**Comment by binwiederhier** [comment]
Whoa. I didn't even know it was live yet. Apple hasn't sent me an email yet that it's live. I would've posted instructions here too. I'll do that later. Right now I'm in IKEA bed building mode.

**Comment by binwiederhier** [comment]
# 📢 🥳  Request for testing

This release of the iOS app adds message priorities (mapped to iOS interruption levels), tags and emojis, action buttons to open websites or perform HTTP requests (in the notification and the detail view), a custom click action when the notification is tapped, and various other fixes.

It also adds support for self-hosted servers (albeit not supporting auth yet). The selfhosted server needs to be configured to forward poll requests to upstream ntfy.sh for push notifications to work (see https;//ntfy.sh/docs/config/#ios-instant-notifications for details).

PLEASE TEST all of the above features. **Report issues ON GITHUB or IN TESTFLIGHT.**

# How to test

**➡️  https://testflight.apple.com/join/P1fFnAm9 ⬅️** 

# Notes
After this release, I will close this issue and I'll be moving towards a business as usual method for iOS releases. I'll still try to catch up with most Android features, but I won't be asking for TestFlight here. Instead, I'll ask in Discord/Matrix, and/or in https://ntfy.sh/announcements.

**Comment by binwiederhier** [comment]
Any positive/negative feedback on the new iOS version?

**Comment by mikniche** [comment]
I've got it working nicely with my self hosted instance of ntfy, after following the instructions and adding the upstream url 🙌🙌

**Comment by robfox92** [comment]
Works well - the only thing that tripped me up was the need to specify the server is using https in the app. I'd initially specified "ntfy.<my_selfhosted_instance>.xyz" instead of "https://ntfy.<my_selfhosted_instance>.xyz". 

Maybe it's worth prepending the server address with https:// automatically?

**Comment by binwiederhier** [comment]
Awesome to hear that it's working. Thanks for the feedback. Keep them coming

> Maybe it's worth prepending the server address with https:// automatically?

:+1: The Android app has validation and a placeholder for this. I'll add something like that.

**Comment by prabirshrestha** [comment]
Tried the selfhosted server and it works great.

iOS Bug:
* Double line to single line for long topics seems to be fixed but now the font sizes changes.

Web Bug:
* If WebSockets is not enabled on the proxy server or fails to connect, it doesn't any error message. I had to open the console and see that there was a websocket error and had to enable WebSocket on the proxy server that comes with synology nas.

iOS feature requests:
* My domain is not small as `notify.sh`. Remember my last used settings for "Use another server" and the "Server URL" or even better if the "Server URL" is a list of predefined servers where user can add or remove.

ntfy feature requests:
* add support for LDAP auth so it plays nicely with rest of the self hosted ecosystem. You can try it with [gldap](https://github.com/jimlambrt/gldap) server. I would like to avoid mantaining yet another user/password combination. This is easier for family members that are non-tech.
* add support for tokens. I would like to use different token with custom expiration.
* add support for making a topic as listed or unlisted similar to youtube https://support.google.com/youtube/answer/3127309?hl=en. This will allow a family member to register a server and then easily discover listed topics.

docs request:
* What happens if `upstream-base-url` is set but fails to send to the upstream server. This could be due to temporary internet connection getting down. Will this be retried or will be lost.
* This is a bit confusing.

>The ntfy.sh server publishes the message to Firebase, which forwards it to APNS, which forwards it to your iOS device
Your iOS device receives the poll request, and fetches the actual message from your server, and then displays it

`The ntfy.sh server publishes the message to Firebase`. What is a message here? Is it just the poll id and topic hash?

**Comment by binwiederhier** [comment]
Thank you for the detailed feedback. I love it :heart: Would you mind reporting bugs and feature requests in individual tickets? It's impossible for me to keep track of things if they are in comments. Thanks.

I'll respond to your comments in detail tomorrow.

**Comment by binwiederhier** [comment]
# Closing this ticket :partying_face: 
With the release of the [iOS v1.1 app](https://apps.apple.com/app/ntfy/id1625396347) :partying_face:, I am ceremonially closing this ticket. :partying_face:  :partying_face:  :partying_face: 

You may of course continue to leave feedback here, or create tickets in the main GitHub.

# Discord/Matrix + ntfy.sh/announcements
I also encourage you to join [Discord](https://discord.gg/cT7ECsZj9w) or [Matrix](https://matrix.to/#/#ntfy:matrix.org), and subscribe to the https://ntfy.sh/announcements topic. I'll potentially be asking for TestFlight testers there.
